src/status.ts

```diff
@@ -84,7 +84,7 @@ export class StatusBar implements vscode.Disposable {
       setVisible(item, this._visible && !!item.text);
     }
     setVisible(this._debugButton,
-               this._visible && vscode.extensions.getExtension('ms-vscode.cpptools') !== undefined
+               this._visible && !this._hideDebugButton && vscode.extensions.getExtension('ms-vscode.cpptools') !== undefined
```

Comment:
I think you need to hide `this._launchTargetNameButton` as well.  What about the ctest button?

Reply:
As explained in the description, launchTargetName could still be useful, because it provides the wanted binary name. So it could be used in launch.json.

ctest isn't relevant since it has nothing to do with the desktop program assumption.